quil-rs/src/program/mod.rs

```diff
@@ -1048,6 +1084,30 @@ I 0
         assert_abs_diff_eq!(matrix.as_ref().unwrap(), expected);
     }
 
+    /// Tests that the various methods of getting the instructions from a Program produce
+    /// consistent results.
+    #[test]
+    fn test_to_instructions() {
```

Comment:
Added some more tests because it's too easy to miss updating some of these methods when adding something like this.